mklive.sh.in

```diff
@@ -337,6 +355,22 @@ if [ "$(id -u)" -ne 0 ]; then
     die "Must be run as root, exiting..."
 fi
 
+#The -d and -t options are complementary. If one exists, the other must also exist. 
+#If these options are set, I also check sbsign command.
+if  ([ -z $DBKEY ] &&  [ ! -z $DBCRT ]) || ([ ! -z $DBKEY ] &&  [ -z $DBCRT ]); then
+    die "Must be set a key and certificate via -d and -t option, exiting..."
+elif ([ $DBKEY ] && [ $DBCRT ]); then
```

Comment:
Ugly! Maybe?
```sh
if [ -n "$key" -a -n "$crt" ]; then
    ...
else if [ -n "$key$crt" ]; then
    die "...."
fi
```

Reply:
-a and -o are deprecated (according to bash faq), -n gives no much value in general if any, and similarly ! can be basically used instead of -z :D

Reply:
I was talking about the order of execution ;)

Reply:
sure, the structure is fine :)